build.gradle

```diff
@@ -125,10 +125,22 @@ processResources {
 }
 
 jar {
-
-    configurations.shade.each { dep ->
-        from(project.zipTree(dep)){
-            exclude 'META-INF', 'META-INF/**'
+    // Shading could try to add duplicate classes
+    // fail instead of silently adding multiple entries with the same name
+    duplicatesStrategy=DuplicatesStrategy.FAIL
+
+    configurations.shade.getResolvedConfiguration().getResolvedArtifacts().each { dep ->
+        def id = dep.getModuleVersion().id
+        if (configurations.shade.dependencies.find { orig -> orig.group == id.group && orig.name == id.name }) {
```

Comment:
so... only shade stuff thats in the MinecraftForge group?

Reply:
That would miss de.oceanlabs stuff.